common/cluster/metadata.go

```diff
@@ -151,3 +171,45 @@ func (m Metadata) ClusterNameForFailoverVersion(failoverVersion int64) string {
 	}
 	return clusterName
 }
+
+// gets the initial failover version for a cluster / domain
+// along with some helpers for a migration - should it be necessary
+func (m Metadata) getInitialFailoverVersion(cluster string, domainName string) int64 {
+	info, ok := m.allClusters[cluster]
+	if !ok {
+		panic(fmt.Sprintf(
+			"Unknown cluster name: %v with given cluster initial failover version map: %v.",
+			cluster,
+			m.allClusters,
+		))
+	}
+
+	// if using the minFailover Version during a cluster config, then return this from config
+	// (assuming it's safe to do so). This is not the normal state of things and intended only
+	// for when migrating versions.
+	usingMinFailoverVersion := m.useMinFailoverVersionOverride(domainName)
+	if usingMinFailoverVersion && info.MinInitialFailoverVersion != nil && *info.MinInitialFailoverVersion > info.InitialFailoverVersion {
```

Comment:
tbh, I don't think `*info.MinInitialFailoverVersion > info.InitialFailoverVersion` is necessary